lib/active_model/serializer/attributes.rb

```diff
@@ -14,10 +14,12 @@ module Attributes
 
         # Return the +attributes+ of +object+ as presented
         # by the serializer.
-        def attributes(requested_attrs = nil, reload = false)
+        def attributes(options = {}, reload = false)
+          requested_attrs = options[:only]
+          excepts = Array(options[:except])
```

Comment:
Except that the serializer already functions as a whitelist

This feature is only useful when you opt in to more data than you want AND
then choose not to use existing features to exclude data. This is really
the job of the scope and instance options along with attrribte if/unless
and multiple serializers. My gut is this is not a primary use case we want
to support or encourage, and it can probably also use the existing 'fields'
option AND this doesn't apply to jsonapi

A good feature pr isn't free unless op will stick around to maintain it.
On Mon, Mar 7, 2016 at 8:44 AM L. Preston Sego III <notifications@github.com>
wrote:

> In lib/active_model/serializer/attributes.rb
> <https://github.com/rails-api/active_model_serializers/pull/1538#discussion_r55211940>
> :
>
> > @@ -14,10 +14,12 @@ module Attributes
> >
> >          # Return the +attributes+ of +object+ as presented
> >          # by the serializer.
> > -        def attributes(requested_attrs = nil, reload = false)
> > +        def attributes(options = {}, reload = false)
> > +          requested_attrs = options[:only]
> > +          excepts = Array(options[:except])
>
> @noahsilas <https://github.com/noahsilas> if you can add a test using
> only/except and if/unless, I'll accept the PR.
>
> I see this being used for excluding attributes more often than including
> them. Such as excluding the created_at and updated_at fields.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails-api/active_model_serializers/pull/1538/files#r55211940>
> .
>